packages/rum/src/domain/record/startFullSnapshots.ts

```diff
@@ -65,13 +66,17 @@ export function startFullSnapshots(
     }
     return records
   }
-  fullSnapshotCallback(takeFullSnapshot())
+
+  fullSnapshotPendingCallback()
+  fullSnapshotReadyCallback(takeFullSnapshot())
+
   let cancelIdleCallback: (() => void) | undefined
   const { unsubscribe } = lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (view) => {
     flushMutations()
     function takeSubsequentFullSnapshot() {
       flushMutations()
-      fullSnapshotCallback(
+      fullSnapshotPendingCallback()
+      fullSnapshotReadyCallback(
```

Comment:
**💬 suggestion:** ‏there is no point in notifying that the snapshot is pending just before it's ready. Move `fullSnapshotPendingCallback()` before `requestIdleCallback(...)`